src/docfx/build/document/RepositoryProvider.cs

```diff
@@ -27,70 +29,93 @@ internal class RepositoryProvider
             LocalizationProvider? localizationProvider = null)
         {
             _docsetPath = docsetPath;
-            _repository = repository;
+            DefaultRepository = repository;
             _packageResolver = packageResolver;
-            _locale = LocalizationUtility.GetLocale(repository);
             _config = config;
             _localizationProvider = localizationProvider;
-            _templateRepository = new Lazy<(string, Repository?)>(GetTemplateRepository);
+
+            if (DefaultRepository != null)
+            {
+                _docsetPathToDefaultRepository = new PathString(Path.GetRelativePath(DefaultRepository.Path, _docsetPath));
+            }
         }
 
         public Repository? GetRepository(FileOrigin origin, PathString? dependencyName = null)
         {
             return GetRepositoryWithDocsetEntry(origin, dependencyName).repository;
         }
 
+        public (Repository? repository, PathString? pathToRepository) GetRepository(FilePath path)
+        {
+            return path.Origin switch
+            {
+                FileOrigin.Default => GetRepository(Path.Combine(_docsetPath, path.Path)),
+                FileOrigin.Fallback when _localizationProvider != null
+                    => (_localizationProvider.GetFallbackRepositoryWithDocsetEntry().fallbackRepository,
+                        _docsetPathToDefaultRepository is null ? null : _docsetPathToDefaultRepository + path.Path),
```

Comment:
can just be `_docsetPathToDefaultRepository`

Reply:
This is to get the relative path of a _file_ to repository, used to calculate GitHub edit URL and such, it is different from `GetRepositoryWithDocsetEntry`

Reply:
I mean `_docsetPathToDefaultRepository is null ? null : _docsetPathToDefaultRepository` can just be `_docsetPathToDefaultRepository`, it is nullable anyway.

Reply:
Now I get it :)

Reply:
:)